pkg/koordlet/metriccache/storage_tables.go

```diff
@@ -17,17 +17,50 @@ limitations under the License.
 package metriccache
 
 import (
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+	"fmt"
 	"time"
 )
 
 const (
 	NodeCPUInfoRecordType = "NodeCPUInfo"
 )
 
+type gpuResourceMetric struct {
+	Minor      int32     // index starting from 0
+	DeviceUUID string    // device UUID
+	SMUtil     float64   // current utilization rate for the device
+	MemoryUsed float64   // used memory on the device, in bytes
+	Timestamp  time.Time `json:"-"`
+}
+
+type GPUMetricsArray []gpuResourceMetric
+
+func (array *GPUMetricsArray) Scan(value interface{}) error {
```

Comment:
can you add a comment about why you need a customized `Scan()` and `Value()`?

Reply:
Gorm customize data type.
https://gorm.io/docs/data_types.html